### DISCUSSION TITLE
ValidTime format?

### DISCUSSION THREAD

**Discussion by acrouse-y** [weather-gov/api#407] [Q&A]
Almost all of the data I use for validTime has been of the format ".../PT9H" where the specific value is valid for n hours (9 in this case).

Yesterday, I received a single occurrence of windSpeed and windGust (one for each) with a validTime formatted as ".../P1DT3H".  Is this intended to mean 1 day + 3 hours?  The reference material on this portion of the tag has not clarified my understanding.

Thank you.

**Comment by netbymatt** [comment]
The durations are defined in ISO 8601
https://en.m.wikipedia.org/wiki/ISO_8601#Durations

**Comment by JJKraw** [comment]
Expanding a bit on the difference between the two...

Anything that appears after the P and before the T are values of a day length or longer (Year Month Week Day). After the T are values less than a day (Hours Minutes Seconds). If the T appears immediately after the P, that means none of the longer elements appear.

So, yes, the correct interpretation of the latter validTime is 1 day 3 hours.

Not sure what you are processing them with, but PHP has a class (DateInterval) that converts ISO 8601 into something easily usable.

**Reply by acrouse-y** [reply]
Thank you for the replies which have clarified the format.  I'm using C++ which has a dateinterval class, but apparently quite different from the PHP class and not useful in this context.

I had been working with XML weather data from a service offered by the Western Regional Office until that service was discontinued in Feb 2021.  The format chosen for the national API has been significantly more 'challenging' to parse and utilize.

Thank you for the kind guidance on this issue.

**Reply by acrouse-y** [reply]
This morning I encountered this temperature data:

{
                    "validTime": "2021-05-13T11:00:00+00:00/PT2H",
                    "value": 13.888888888888889
                },
                {
                    "validTime": "2021-05-13T13:00:13+00:00**_/PT59M47S"_**,
                    "value": 15
                },
                {
                    "validTime": "2021-05-13T14:00:00+00:00/PT1H",
                    "value": 16.666666666666668
                },

Valid in terms of the ISO standard, but logically inconsistent.
